Сделал дз  к 2 уроку

hmzxpDev opened the pull request.

hmzxpDev committed:
Сделал дз  к 2 уроку

SokolAlexander commented on src/App.js:
```diff
@@ -0,0 +1,70 @@
+        <Message messageList={messageList}></Message>
+      </div>
+      {/* форма отправки */}
+      <div>
```
Хорошо бы завернуть в form и использовать onSubmit

SokolAlexander reviewed (approved):
В целом ок) небольшой коммент

hmzxpDev committed:
сделал 3 задание

SokolAlexander commented on src/App.js:
```diff
@@ -1,12 +1,35 @@
+  const [chatList, setContactList] = useState([{ id: 1, name: 'Игорь' }, { id: 2, name: 'Тима' }, { id: 3, name: 'Сергей' }])
+
+  // Стилизация компонентов
+  const MyButton = styled(Button)({
```
это нужно делать вне компонентов

SokolAlexander reviewed (approved):
Оставил коммент, также не нашел фокуса через реф, в остальном ок

Commit:
Сделал дз к 4 уроку

1. Я решил сделать через перебор чата. На уроке шла речь о плохой оптимизации, но я решил не запариваться над этим, потому что это не относиться к курсу реакт. А сидеть и пытаться оптимизировать чат(когда в реальных условиях данные поступают с бека) считаю не логичным.

2. если пользователь вводит идентификатор несуществующего чата или если идентификатора чата нет :
программа не падает, потому что построена не как у вас. И я посмотрел в вк, если введи в url несуществующий id, он просто выдаст открытый чат с delete аккаунтом.

SokolAlexander commented on src/components/Chat/index.js:
```diff
@@ -0,0 +1,77 @@
+    const [chatList] = useState(Chats)
+    const { ChatId } = useParams(); // динамические данные
+    // стейт специально для сохранения состояния и ответа робота
+    const [params, setParams] = useState(ChatId)
```
интересное решение, но в этом нет необходимости)

SokolAlexander reviewed (approved):
Гуд
